features/manage-position/MethodSelector.tsx

```diff
@@ -40,44 +42,70 @@ interface IProps {
 }
 
 const MethodSelector = ({ method, handleChange }: IProps) => {
-  return (
-    <Box py={2}>
-      <FormWrapper>
-        <InputLabel id="demo-simple-select-label">Actions</InputLabel>
-        <Select
-          labelId="demo-simple-select-label"
-          value={method}
-          onChange={handleChange}
-          input={<BootstrapInput />}
-        >
-          <MenuItem value={"create"}>
-            <ListItemText
-              primary="Create"
-              secondary="Mint new synthetic tokens."
-            />
-          </MenuItem>
-          <MenuItem value={"deposit"}>
-            <ListItemText
-              primary="Deposit"
-              secondary="Add to position collateral."
-            />
-          </MenuItem>
-          <MenuItem value={"withdraw"}>
-            <ListItemText
-              primary="Withdraw"
-              secondary="Remove position collateral"
-            />
-          </MenuItem>
-          <MenuItem value={"redeem"}>
-            <ListItemText
-              primary="Redeem"
-              secondary="Redeem synthetics for collateral."
-            />
-          </MenuItem>
-        </Select>
-      </FormWrapper>
-    </Box>
-  );
+  const { empState } = EmpState.useContainer();
+  const { isExpired } = empState;
+  if (isExpired !== null) {
+    return renderComponent(isExpired);
+  } else {
+    return renderComponent();
+  }
+
+  function renderComponent(_isExpired: boolean = false) {
+    return (
+      <Box py={2}>
+        <FormWrapper>
+          <InputLabel id="demo-simple-select-label">Actions</InputLabel>
+          <Select
+            labelId="demo-simple-select-label"
+            value={method}
+            onChange={handleChange}
+            input={<BootstrapInput />}
+          >
+            {_isExpired && (
+              <MenuItem value={"settle"}>
+                <ListItemText
+                  primary="Settle"
+                  secondary="Settle expired tokens."
+                />
+              </MenuItem>
+            )}
+            {!_isExpired && (
```

Comment:
could you not wrap all the items below this one that you don't want to render when `!_isExpired` in a `<div> ... </div>` and then only query this `!_isExpired` once?

Reply:
You're right!